src/CgiRequestHandler.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "CgiHandler.hpp"
+#include "Config.hpp"
+#include "Header.hpp"
+#include "IRequestHandler.hpp"
+#include "IRequestManager.hpp"
+
+/*
+	Bridge between the Server logic and the CGI process.
+	Implements flow control (backpressure) and timeout management.
+*/
+class CgiRequestHandler : public IRequestHandler
+{
+public:
+	CgiRequestHandler(IRequestManager &manager, const Header &header, const RouteConfig &route);
+	virtual ~CgiRequestHandler();
+
+	void onBodyData(std::span<const char> data) override;
+	void notifyResponseBuffer(size_t bufferSize) override;
+
+	// Should be called periodically by the main loop to detect stuck scripts.
+	void checkTimeout();
+
+private:
+	std::string findInterpreter(const std::string &scriptPath, const RouteConfig &route);
+
+	void startCgiOutputRead();
+
+	IRequestManager &manager_;
+	std::unique_ptr<CgiHandler> cgiHandler_;
```

Comment:
`cgiHandler_` is assigned in constructor and never removed, it could be a direct member instead of `unique_ptr`